### PR TITLE
Add packagist webhook

### PR DESCRIPTION
**Pull request by toddy15**
Hi,

this PR adds a new webhook for [Packagist](https://packagist.org) integration.

Packagist is the main [Composer](https://getcomposer.org/) repository. It aggregates public PHP packages installable with Composer. 

Using the webhook in a repository, a PHP package hosted on a gitea server will automatically update its information on packagist.org on push events.

I've built gitea on my machine and tested the new webhook locally. Everything seems to work as expected.

Regards,
Tobias

### CODE REVIEW

**Comment by S1SYPHOS** [comment]
I've been waiting for this for quite some time, great work, thank you so much!

File: models/migrations/v161.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ func convertTaskTypeToString(x *xorm.Engine) error {
 		FEISHU
 		MATRIX
 		WECHATWORK
+		PACKAGIST
```
Comment (wxiaoguang) [inline models/migrations/v161.go]: Here and below and v162.go, the `PACKAGIST` doesn't need to be processed. At that time we did not have it yet.

**Review by wxiaoguang — APPROVED** [review]

File: models/migrations/v161.go (hunk 1 above)
Comment (toddy15) [reply models/migrations/v161.go]: Thanks for your review and the approval. I've removed the migrations from v161.go and v162.go.

I had already suspected that those migrations are not necessary, but I've looked at the PR for the last webhook addition (wechatnetwork, #15910). In that PR, those migrations have been added. AFAIU, the migrations for WeChat were not necessary either, right?

I've also merged in the latest changes from the gitea main branch and squashed the commits, to keep the commit for this PR as minimal as possible. If you'd like to have separate commits, please say so, and I'll force-push again.

Comment (wxiaoguang) [reply models/migrations/v161.go]: Oh yep, adding or not are both fine .... just small nit, nvm. I didn't review the wechatnetwork PR  #15910  😂

**Review by Gusted — APPROVED** [review]
Once this in the a released stable version we might want to add a notice to the packagist site: https://github.com/composer/packagist https://packagist.org/about under `How to update packages?`

**Comment by toddy15** [comment]
Thanks for your approval. Yes, adding instructions for gitea on the packagist.org website is definitely useful. I'll try to remember sending them a PR when gitea has been released with this webhook. :smile:

**Comment by toddy15** [comment]
@wxiaoguang Should I squash the commits again, rebase and force-push my branch?

**Comment by wxiaoguang** [comment]
> @wxiaoguang Should I squash the commits again, rebase and force-push my branch?

No need. Squash merge works well in GitHub, just wait for the CI to pass 😊

**Pull request merged**